src/extension/background-script/actions/ln/request.ts

```diff
@@ -0,0 +1,82 @@
+import utils from "~/common/lib/utils";
+import { MessageGenericRequest } from "~/types";
+
+import db from "../../db";
+import state from "../../state";
+
+const request = async (
+  message: MessageGenericRequest
+): Promise<{ data: unknown } | { error: string }> => {
+  const connector = await state.getState().getConnector();
+
+  const { origin, args } = message;
+
+  try {
+    if (!connector.requestMethod) {
+      throw new Error(
+        `${message.args.method} is not supported by your account`
+      );
+    }
+
+    const allowance = await db.allowances
```

Comment:
> Why do we check for the allowance here again? This check seems to be duplicate to the general `enabled` checks that happen before even getting into this method?

This is a TypeGuard for Typescript in the first place, because `allowance.id` is needed later to save the permission in the DB (and can be undefined). 
I will keep it for now, as this check doesn't hurt either.

Reply:
> If a user currently deletes an allowance, what happens with the permissions? Should they they be deleted as well? 
> I'd expect to be prompted for permissions again after I deleted an allowance.

Yes, I asked this question is the former PR [here](https://github.com/getAlby/lightning-browser-extension/pull/1695/files#diff-536b7f7cb4ac5b988af6ca70ed282bed5b563cb942b9d242da9f19061ae84d79R10-R12) too 
which got lost on its way to here. 
Thanks for bringing this up again.

I added the corresponding deletion now.